README.md
# Ovalizer - for Sketch

> Apply Oval conventions automatically 🎩.

🚧  WIP 🚧

## Install

Download the [latest version](https://github.com/OvalMoney/ovalizer-sketch/releases) ZIP and double click on `ovalizer.sketchplugin`.

## License

Apache 2.0 © [Oval Money]()


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="0.0.3"></a>
## [0.0.3](https://github.com/OvalMoney/ovalizer-sketch/compare/v0.0.2...v0.0.3) (2019-01-25)


### Bug Fixes

* check-artboards both ITA and ENG ([c0dd977](https://github.com/OvalMoney/ovalizer-sketch/commit/c0dd977))



<a name="0.0.2"></a>
## [0.0.2](https://github.com/OvalMoney/ovalizer-sketch/compare/v0.0.1...v0.0.2) (2019-01-25)


### Bug Fixes

* artboard check filter ([7650793](https://github.com/OvalMoney/ovalizer-sketch/commit/7650793))



<a name="0.0.1"></a>
## 0.0.1 (2019-01-25)


src/lib/artboards/autoalign.js
import rowsByPosition from './rows-by-position';

export default function autoalign(page) {
	const rows = rowsByPosition(page);

	const rowNames = Object.keys(rows).sort(
		(a, b) => parseInt(a, 10) - parseInt(b, 10)
	);

	let y = 0;
	for (const rowName of rowNames) {
		const row = rows[rowName];
		let x = 0;
		let nextYOffset = 1000;
		let sequenceNumber = 0;

		for (const artboard of row) {
			// Make sure they're in the right order, so the list on the left is sorted
			const parent = artboard.parentGroup();
			artboard.removeFromParent();
			parent.insertLayer_atIndex(artboard, 0);

			// Update name
			const artboardNumber = parseInt(rowName, 10) + sequenceNumber;
			artboard.name = artboardNumber.toString();

			// Update artboard's position
			// artboard.frame().{x,y}() isn't always relatively to (0,0), and using
			// absoluteRect.ruler{X,Y} seems to solve this
			artboard.absoluteRect().rulerX = x;
			artboard.absoluteRect().rulerY = y;

			// Use the height of the largest artboard on this row to determine the
			// y-offset of the next row (plus a small buffer for labels)
			const height = artboard.frame().height() + 100;
			if (height > nextYOffset) {
				// Snap to a 500 unit grid
				nextYOffset = height + (500 - (height % 500));
			}

			x += 500;
			sequenceNumber++;
		}
		y += nextYOffset;
	}
}


src/lib/artboards/check-artboards.js
import rowsByPosition from './rows-by-position';

const getArtboards = rows => Object.keys(rows).map(rowName => rows[rowName]);
const filterArtboards = (artboardToFilter, invalidArtboards) =>
	artboardToFilter.filter(artboard => !invalidArtboards.find(invalid => invalid.name() === artboard.name()));

export default function checkArtoboards(pageIt, pageEn) {
	const rowsIt = rowsByPosition(pageIt);
	const rowsEn = rowsByPosition(pageEn);

	const artboardsItRows = getArtboards(rowsIt);
	const artboardsEnRows = getArtboards(rowsEn);
	const innerArtboardsIt = artboardsItRows.reduce((acc, artboardGroup) => acc.concat(...artboardGroup), []);
	const innerArtboardsEn = artboardsEnRows.reduce((acc, artboardGroup) => acc.concat(...artboardGroup), []);

	if (innerArtboardsEn.length !== innerArtboardsIt.length) {
		// If ENG is invalid filter on IT to find what's missing
		const missingArtboardsEn = filterArtboards(innerArtboardsIt, innerArtboardsEn).map(artboard => artboard.name());
		const missingArtboardsIt = filterArtboards(innerArtboardsEn, innerArtboardsIt).map(artboard => artboard.name());

		const missingItaMessage = missingArtboardsIt.length > 0 ? `ITA: ${missingArtboardsIt} ` : '';
		const missingEngMessage = missingArtboardsEn.length > 0 ? `ENG: ${missingArtboardsEn}` : '';
		return {message: '⚠️ Artboards missing. Details: '.concat(missingItaMessage).concat(missingEngMessage)};
	}
}


src/lib/artboards/rows-by-position.js

export default function rowsByPosition(page) {
	const artboards = page.artboards();

	// Find the row covers
	const rowCovers = artboards.reduce((acc, artboard) => {
		const number = parseInt(artboard.name(), 10);
		return number % 100 === 0 ? acc.concat(artboard) : acc;
	}, []);

	// Sort the existing covers
	rowCovers.sort((a, b) => a.frame().y() - b.frame().y());

	// Build up the rows, starting with their covers
	const rows = rowCovers.reduce((acc, row, index) => {
		const rowName = ((index + 1) * 100).toString();
		acc[rowName] = [row];
		return acc;
	}, {});

	const yPositions = Object.entries(rows).map(([name, row]) => [
		name,
		row[0].frame().y()
	]);

	// Add other artboards into the rows
	for (let i = 0; i < artboards.length; i++) {
		// Skip row covers, obvs...
		if (rowCovers.includes(artboards[i])) {
			continue;
		}

		const yPos = artboards[i].frame().y();

		let closestRow;
		let smallestDist = Infinity;
		for (const [rowName, rowYPos] of yPositions) {
			const yDist = Math.abs(yPos - rowYPos);
			if (yDist < smallestDist) {
				closestRow = rowName;
				smallestDist = yDist;
			}
		}

		rows[closestRow].push(artboards[i]);
	}

	// Sort rows by board x position
	Object.values(rows).forEach(row =>
		row.sort((a, b) => a.frame().x() - b.frame().x())
	);

	return rows;
}


ovalizer.sketchplugin/Contents/Sketch/lint.js
var that = this;
function __skpm_run (key, context) {
  that.context = context;

var exports =
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = "";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = "./src/commands/lint.js");
/******/ })
/************************************************************************/
/******/ ({

/***/ "./src/commands/lint.js":
/*!******************************!*\
  !*** ./src/commands/lint.js ***!
  \******************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "default", function() { return validateAndFix; });
/* harmony import */ var sketch_ui__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! sketch/ui */ "sketch/ui");
/* harmony import */ var sketch_ui__WEBPACK_IMPORTED_MODULE_0___default = /*#__PURE__*/__webpack_require__.n(sketch_ui__WEBPACK_IMPORTED_MODULE_0__);
/* harmony import */ var _lib_pages__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ../lib/pages */ "./src/lib/pages/index.js");
/* harmony import */ var _lib_artboards_autoalign__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ../lib/artboards/autoalign */ "./src/lib/artboards/autoalign.js");
/* harmony import */ var _lib_artboards_fix_title__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! ../lib/artboards/fix-title */ "./src/lib/artboards/fix-title.js");
/* harmony import */ var _lib_artboards_check_artboards__WEBPACK_IMPORTED_MODULE_4__ = __webpack_require__(/*! ../lib/artboards/check-artboards */ "./src/lib/artboards/check-artboards.js");
/* eslint-disable-next-line import/no-unresolved */





function validateAndFix(context) {
  var _getPages = Object(_lib_pages__WEBPACK_IMPORTED_MODULE_1__["getPages"])(context),
      itPage = _getPages.itPage,
      enPage = _getPages.enPage;

  if (!itPage && !enPage) {
    sketch_ui__WEBPACK_IMPORTED_MODULE_0___default.a.message('⚠️ Missing ITA or ENG page');
    return;
  }

  if (itPage) {
    Object(_lib_artboards_autoalign__WEBPACK_IMPORTED_MODULE_2__["default"])(itPage);
    Object(_lib_artboards_fix_title__WEBPACK_IMPORTED_MODULE_3__["default"])(itPage);
  }

  if (enPage) {
    Object(_lib_artboards_autoalign__WEBPACK_IMPORTED_MODULE_2__["default"])(enPage);
    Object(_lib_artboards_fix_title__WEBPACK_IMPORTED_MODULE_3__["default"])(enPage);
  }

  if (itPage && enPage) {
    var _checkArtboards = Object(_lib_artboards_check_artboards__WEBPACK_IMPORTED_MODULE_4__["default"])(itPage, enPage),
        message = _checkArtboards.message;

    sketch_ui__WEBPACK_IMPORTED_MODULE_0___default.a.message(message);
  }
}

/***/ }),

/***/ "./src/lib/artboards/autoalign.js":
/*!****************************************!*\
  !*** ./src/lib/artboards/autoalign.js ***!
  \****************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "default", function() { return autoalign; });
/* harmony import */ var _rows_by_position__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./rows-by-position */ "./src/lib/artboards/rows-by-position.js");

function autoalign(page) {
  var rows = Object(_rows_by_position__WEBPACK_IMPORTED_MODULE_0__["default"])(page);
  var rowNames = Object.keys(rows).sort(function (a, b) {
    return parseInt(a, 10) - parseInt(b, 10);
  });
  var y = 0;
  var _iteratorNormalCompletion = true;
  var _didIteratorError = false;
  var _iteratorError = undefined;

  try {
    for (var _iterator = rowNames[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
      var rowName = _step.value;
      var row = rows[rowName];
      var x = 0;
      var nextYOffset = 1000;
      var sequenceNumber = 0;
      var _iteratorNormalCompletion2 = true;
      var _didIteratorError2 = false;
      var _iteratorError2 = undefined;

      try {
        for (var _iterator2 = row[Symbol.iterator](), _step2; !(_iteratorNormalCompletion2 = (_step2 = _iterator2.next()).done); _iteratorNormalCompletion2 = true) {
          var artboard = _step2.value;
          // Make sure they're in the right order, so the list on the left is sorted
          var parent = artboard.parentGroup();
          artboard.removeFromParent();
          parent.insertLayer_atIndex(artboard, 0); // Update name

          var artboardNumber = parseInt(rowName, 10) + sequenceNumber;
          artboard.name = artboardNumber.toString(); // Update artboard's position
          // artboard.frame().{x,y}() isn't always relatively to (0,0), and using
          // absoluteRect.ruler{X,Y} seems to solve this

          artboard.absoluteRect().rulerX = x;
          artboard.absoluteRect().rulerY = y; // Use the height of the largest artboard on this row to determine the
          // y-offset of the next row (plus a small buffer for labels)

          var height = artboard.frame().height() + 100;

          if (height > nextYOffset) {
            // Snap to a 500 unit grid
            nextYOffset = height + (500 - height % 500);
          }

          x += 500;
          sequenceNumber++;
        }
      } catch (err) {
        _didIteratorError2 = true;
        _iteratorError2 = err;
      } finally {
        try {
          if (!_iteratorNormalCompletion2 && _iterator2.return != null) {
            _iterator2.return();
          }
        } finally {
          if (_didIteratorError2) {
            throw _iteratorError2;
          }
        }
      }

      y += nextYOffset;
    }
  } catch (err) {
    _didIteratorError = true;
    _iteratorError = err;
  } finally {
    try {
      if (!_iteratorNormalCompletion && _iterator.return != null) {
        _iterator.return();
      }
    } finally {
      if (_didIteratorError) {
        throw _iteratorError;
      }
    }
  }
}

/***/ }),

/***/ "./src/lib/artboards/check-artboards.js":
/*!**********************************************!*\
  !*** ./src/lib/artboards/check-artboards.js ***!
  \**********************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "default", function() { return checkArtoboards; });
/* harmony import */ var _rows_by_position__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./rows-by-position */ "./src/lib/artboards/rows-by-position.js");
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }



var getArtboards = function getArtboards(rows) {
  return Object.keys(rows).map(function (rowName) {
    return rows[rowName];
  });
};

var filterArtboards = function filterArtboards(artboardToFilter, invalidArtboards) {
  return artboardToFilter.filter(function (artboard) {
    return !invalidArtboards.find(function (invalid) {
      return invalid.name() === artboard.name();
    });
  });
};

function checkArtoboards(pageIt, pageEn) {
  var rowsIt = Object(_rows_by_position__WEBPACK_IMPORTED_MODULE_0__["default"])(pageIt);
  var rowsEn = Object(_rows_by_position__WEBPACK_IMPORTED_MODULE_0__["default"])(pageEn);
  var artboardsItRows = getArtboards(rowsIt);
  var artboardsEnRows = getArtboards(rowsEn);
  var innerArtboardsIt = artboardsItRows.reduce(function (acc, artboardGroup) {
    return acc.concat.apply(acc, _toConsumableArray(artboardGroup));
  }, []);
  var innerArtboardsEn = artboardsEnRows.reduce(function (acc, artboardGroup) {
    return acc.concat.apply(acc, _toConsumableArray(artboardGroup));
  }, []);

  if (innerArtboardsEn.length !== innerArtboardsIt.length) {
    // If ENG is invalid filter on IT to find what's missing
    var missingArtboardsEn = filterArtboards(innerArtboardsIt, innerArtboardsEn).map(function (artboard) {
      return artboard.name();
    });
    var missingArtboardsIt = filterArtboards(innerArtboardsEn, innerArtboardsIt).map(function (artboard) {
      return artboard.name();
    });
    var missingItaMessage = missingArtboardsIt.length > 0 ? "ITA: ".concat(missingArtboardsIt, " ") : '';
    var missingEngMessage = missingArtboardsEn.length > 0 ? "ENG: ".concat(missingArtboardsEn) : '';
    return {
      message: '⚠️ Artboards missing. Details: '.concat(missingItaMessage).concat(missingEngMessage)
    };
  }
}

/***/ }),

/***/ "./src/lib/artboards/fix-title.js":
/*!****************************************!*\
  !*** ./src/lib/artboards/fix-title.js ***!
  \****************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "default", function() { return autoalign; });
/* harmony import */ var _rows_by_position__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./rows-by-position */ "./src/lib/artboards/rows-by-position.js");

var placeholder = '{{artboardName}}';
function autoalign(page) {
  var rows = Object(_rows_by_position__WEBPACK_IMPORTED_MODULE_0__["default"])(page);
  var rowNames = Object.keys(rows);

  for (var _i = 0; _i < rowNames.length; _i++) {
    var rowName = rowNames[_i];
    var row = rows[rowName];
    var _iteratorNormalCompletion = true;
    var _didIteratorError = false;
    var _iteratorError = undefined;

    try {
      var _loop = function _loop() {
        var artboard = _step.value;
        var layers = artboard.layers();
        layers.forEach(function (item) {
          if (String(item.name()) === placeholder) {
            item.setStringValue(artboard.name());
          }
        });
      };

      for (var _iterator = row[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
        _loop();
      }
    } catch (err) {
      _didIteratorError = true;
      _iteratorError = err;
    } finally {
      try {
        if (!_iteratorNormalCompletion && _iterator.return != null) {
          _iterator.return();
        }
      } finally {
        if (_didIteratorError) {
          throw _iteratorError;
        }
      }
    }
  }
}

/***/ }),

/***/ "./src/lib/artboards/rows-by-position.js":
/*!***********************************************!*\
  !*** ./src/lib/artboards/rows-by-position.js ***!
  \***********************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "default", function() { return rowsByPosition; });
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function rowsByPosition(page) {
  var artboards = page.artboards(); // Find the row covers

  var rowCovers = artboards.reduce(function (acc, artboard) {
    var number = parseInt(artboard.name(), 10);
    return number % 100 === 0 ? acc.concat(artboard) : acc;
  }, []); // Sort the existing covers

  rowCovers.sort(function (a, b) {
    return a.frame().y() - b.frame().y();
  }); // Build up the rows, starting with their covers

  var rows = rowCovers.reduce(function (acc, row, index) {
    var rowName = ((index + 1) * 100).toString();
    acc[rowName] = [row];
    return acc;
  }, {});
  var yPositions = Object.entries(rows).map(function (_ref) {
    var _ref2 = _slicedToArray(_ref, 2),
        name = _ref2[0],
        row = _ref2[1];

    return [name, row[0].frame().y()];
  }); // Add other artboards into the rows

  for (var i = 0; i < artboards.length; i++) {
    // Skip row covers, obvs...
    if (rowCovers.includes(artboards[i])) {
      continue;
    }

    var yPos = artboards[i].frame().y();
    var closestRow = void 0;
    var smallestDist = Infinity;
    var _iteratorNormalCompletion = true;
    var _didIteratorError = false;
    var _iteratorError = undefined;

    try {
      for (var _iterator = yPositions[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
        var _step$value = _slicedToArray(_step.value, 2),
            rowName = _step$value[0],
            rowYPos = _step$value[1];

        var yDist = Math.abs(yPos - rowYPos);

        if (yDist < smallestDist) {
          closestRow = rowName;
          smallestDist = yDist;
        }
      }
    } catch (err) {
      _didIteratorError = true;
      _iteratorError = err;
    } finally {
      try {
        if (!_iteratorNormalCompletion && _iterator.return != null) {
          _iterator.return();
        }
      } finally {
        if (_didIteratorError) {
          throw _iteratorError;
        }
      }
    }

    rows[closestRow].push(artboards[i]);
  } // Sort rows by board x position


  Object.values(rows).forEach(function (row) {
    return row.sort(function (a, b) {
      return a.frame().x() - b.frame().x();
    });
  });
  return rows;
}

/***/ }),

/***/ "./src/lib/pages/index.js":
/*!********************************!*\
  !*** ./src/lib/pages/index.js ***!
  \********************************/
/*! exports provided: getPages */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

"use strict";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, "getPages", function() { return getPages; });
function getPages(context) {
  var pages = context.document.pages();
  var itPage = pages.find(function (page) {
    return String(page.name).includes('ITA');
  });
  var enPage = pages.find(function (page) {
    return String(page.name).includes('ENG');
  });
  return {
    itPage: itPage,
    enPage: enPage
  };
}

/***/ }),

/***/ "sketch/ui":
/*!****************************!*\
  !*** external "sketch/ui" ***!
  \****************************/
/*! no static exports found */
/***/ (function(module, exports) {

module.exports = require("sketch/ui");

/***/ })

/******/ });
  if (key === 'default' && typeof exports === 'function') {
    exports(context);
  } else {
    exports[key](context);
  }
}
that['onRun'] = __skpm_run.bind(this, 'default')

//# sourceMappingURL=lint.js.map